packages/pigeon/README.md

```diff
@@ -1,13 +1,354 @@
 # Pigeon
 
+// TODO: how to frame this?
+// To get started, see the Quickstart Guide in the [Example README](./example/README.md).
+
 Pigeon is a code generator tool to make communication between Flutter and the
 host platform type-safe, easier, and faster.
 
-Pigeon removes the necessity to manage strings across multiple platforms and languages.
-It also improves efficiency over common method channel patterns. Most importantly though,
-it removes the need to write custom platform channel code, since pigeon generates it for you.
+Pigeon works by reading special files which by convention are placed in a separate
+folder, `/pigeons`, alongside your `/lib` folder. You can put all of your definitions
+in a single file (e.g., `/pigeons/messages.dart`), or you can divide your methods
+and classes into separate files as you wish. After reading these files, Pigeon
+generates:
+
+ * New Dart source files within your `/lib` folder at the specific paths you
+specify, and
+ * New native source files with matching code within your native projects at the
+ specific paths you specify
+
+Internally, the generated code uses `MethodChannel`s to communicate across Flutter's
+UI thread and the Platform thread where your host app initially launched. The value
+in Pigeon comes from automatically keeping this unpleasant boilerplate in sync and
+efficiently marshalling data between languages.
+
+The generated code can either flow from Dart to native code, or native code back
+to Dart. Generated code on the receiving end uses interfaces or abstract classes,
+allowing you to provide implementations in concrete classes.
+
+Pigeon works in both complete Flutter apps and hybrid apps using the add-to-app paradigm.
+
+## Quickstart
+
+### Installation
+
+Begin by adding Pigeon to your Dart project's `pubspec.yaml` file:
+
+```sh
+$ flutter pub add pigeon --dev
+$ flutter pub get
+```
+
+### Setup
+
+To specify what code Pigeon should generate, create a `/pigeons` directory and add
+a `.dart` configuration file. This guide will place all such configurations within
+`/pigeons/messages.dart`, but you are free to choose any file name(s) you like.
+
+Begin by instantiating a `PigeonOptions` object, wrapped in the `@ConfigurePigeon`
```

Comment:
This seems like a long-form tutorial rather than a quick-start.

Maybe a codelab would be a better place for some of this content?